logger.go

```diff
@@ -1,41 +0,0 @@
-package shell
-
-import (
-	"context"
-	"encoding/json"
-	"io"
-)
-
-// Logger is used to handle incoming logs from the ipfs node
-type Logger struct {
-	resp io.ReadCloser
-	dec  *json.Decoder
-}
-
-// Next is used to retrieve the next event from the logging system
-func (l Logger) Next() (map[string]interface{}, error) {
-	var out map[string]interface{}
-	return out, l.dec.Decode(&out)
-}
-
-// Close is used to close our reader
-func (l Logger) Close() error {
-	return l.resp.Close()
-}
-
-// GetLogs is used to retrieve a parsable logger object
-func (s *Shell) GetLogs(ctx context.Context) (Logger, error) {
-	resp, err := s.Request("log/tail").Send(ctx)
-	if err != nil {
-		return Logger{}, err
-	}
-	if resp.Error != nil {
-		resp.Output.Close()
-		return Logger{}, resp.Error
-	}
-	return newLogger(resp.Output), nil
-}
```

Comment:
Does `log tail` not produce anything anymore such that it's worth removing the command here?

e.g. is this shareness test going to fail as well https://github.com/ipfs/go-ipfs/blob/ef866a1400b3b2861e5e8b6cc9edc8633b890a0a/test/sharness/t0065-active-requests.sh#L23?

Reply:
If that log output was coming from go-libp2p, it would have disappeared with v0.18. If it was coming from an ipfs component that hasn't been updated to go-log@v2, we probably need to rework this sharness test.

Reply:
I'm not sure how many/which things within go-ipfs are using go-logv1 still. However, some local cursory checking of go-ipfs shows that there's at least https://github.com/ipfs/go-ipfs/blob/ef866a1400b3b2861e5e8b6cc9edc8633b890a0a/core/corehttp/logs.go#L53 which means that `curl http://localhost:5001/logs` triggers `{"event":"log API client connected","session":"ac177dbb-08e8-4d58-a69b-c0d3d46f6aa7","system":"core/server","time":"2022-03-01T09:36:38.577943Z"}` to show up from `ipfs log tail`.

So we should probably restore the command and if we want we can restore the test as well as long as it does that HTTP call also to trigger an event.

I've also filed https://github.com/ipfs/go-ipfs/issues/8753 since go-ipfs probably wants to deal with the upgrade and fixing `ipfs log tail` sooner rather than later.